tests/test_telemetry.py

```diff
@@ -0,0 +1,21 @@
+from datachain.lib.dc import DataChain
+from datachain.telemetry import is_enabled
+
+
+def test_is_enabled():
+    assert not is_enabled()
+
+
+def test_telemetry_api_call(mocker, tmp_dir):
+    patch_send = mocker.patch("iterative_telemetry.IterativeTelemetryLogger.send")
+
+    DataChain.from_storage(tmp_dir.as_uri())
```

Comment:
[Q] What will be sent for a chain for the form?

`DataChain.from_storage(tmp_dir.as_uri()).map(...).show()` ? Will we get three events and will there be a way to link these events together into a "user session"?

Reply:
Updated the logic. PTAL.

Reply:
tl;dr - I appreciate the effort required to implement proper telemetry is high and this is more than likely good enough.

FWIW - This seems okay for a first pass, but it will not give us any information about how the product is being used. This should however be enough for the active loyal users metric which is probably all we care about as we currently do not have a person dedicated to looking at telemetry/making decisions from it. To give an idea of usage I think we would need to send an event as/when each chain is evaluated (e.g. `save`, `exec`, `to_records`, `collect`, etc).

Reply:
I will be looking at these metrics monthly as I have done every month for the last 3 years and 7 months.  That nobody pays attention in the company has nothing to do with the fact that someone is dedicated to this and is extremely frustrated that we do not do this better and make better decisions from this data. 

Questions/comments:
1. Is the loyal monthly users as being set up here now equal to how it is applied for DVC (5 times usage or more per month)?
2. I agree that as much information in the chain  as possible would be highly desired from the beginning - we will be able to see where people get stuck.  Otherwise we could just be counting trials. with no idea of next steps.  This will be of minimal value like the data from the VS Code extension.
3. It makes more sense to do it right from the beginning so that we can have good data. 
4. Adding for a wholistic view: Our new use of Reo.dev will help to some extent in determining user journey (what people use, where they get stuck, etc) because it will be able to track who and from what company hits our doc pages and when (😃 🎉). But it would be immensely better to know this from the product telemetry.

Reply:
> This will be of minimal value like the data from the VS Code extension. 

qq - did you get / have access to Azure ? that's where all real product analytics for VS Code is (and it's very comprehensive).

we also have the breakdown on the DVC usage (anonymized, etc) - which commands are being used more, etc. 

And this is not about someone not paying attention (we did with @mattseddon and @dberenbaum ). I think it boils down to the fact that you need to get to some different numbers to start benefit from more granular events in a meaningful way.

On the other hand it can potentially slow down the product a bit, a risk of breaking the trust with the community, etc. My 2cs - I would start with basics and expand if needed later.

Reply:
> On the other hand it can potentially slow down the product a bit, a risk of breaking the trust with the community, etc. My 2cs - I would start with basics and expand if needed later.

I agree. IMO, let's merge this and look at the metrics to see if the information we get are enough or not. If we need more information, we can then revive `api_telemetry` decorator and attach it to few function we want to look at like save, exec, to_records and so on.